lib/src/components/NxGlobalSidebar/__tests__/NxGlobalSidebarFooter.test.tsx

```diff
@@ -4,63 +4,61 @@
+
+  it('passes additional classes specified as props to <footer>', function() {
+    const el = renderEl({ className: 'test-class' }),
+        defaultEl = renderEl()!;
+
+    expect(el).toHaveClass('test-class');
 
-  it('renders an <footer> with nx-global-sidebar__footer class and those passed as props', function() {
-    expect(getShallowComponent({ className: 'test-class' }))
-        .toMatchSelector('footer.nx-global-sidebar__footer.test-class');
+    for (const cls of Array.from(defaultEl.classList)) {
+      expect(el).toHaveClass(cls);
+    }
   });
 
   it('passes additional attributes to <footer>', function() {
-    expect(getShallowComponent({ id: 'test-id' })).toHaveProp('id', 'test-id');
-    expect(getShallowComponent({ lang: 'en-ca' })).toHaveProp('lang', 'en-ca');
+    expect(renderEl({ id: 'test-id' })).toHaveAttribute('id', 'test-id');
+    expect(renderEl({ lang: 'en-ca' })).toHaveAttribute('lang', 'en-ca');
   });
 
   it('renders the Support div with a link with the passed href and text and does not render when empty', function() {
-    const supportEmpty = getShallowComponent().find('div.nx-global-sidebar__support'),
-        support = getShallowComponent({ supportText: 'Support for RSC', supportLink: '#supporturl' })
-            .find('div.nx-global-sidebar__support'),
-        supportLink = support.find(NxTextLink);
+    const { container: viewWithoutSupportText} = quickRender(),
+        viewWithSupportText = quickRender({ supportText: 'Support for RSC', supportLink: '#supporturl' }),
+        supportLink = viewWithSupportText.getByRole('link');
 
-    expect(supportEmpty).not.toExist();
-    expect(support).toContainMatchingElement('span.nx-global-sidebar__support-text');
-    expect(support.find('span.nx-global-sidebar__support-text')).toHaveText('Support for RSC');
-    expect(supportLink).toHaveProp('href', '#supporturl');
+    expect(viewWithSupportText.getByText('Support for RSC')).toBeVisible();
+    expect(supportLink).toHaveAttribute('href', '#supporturl');
+
+    expect(viewWithoutSupportText.querySelector('.nx-global-sidebar__support')).not.toBeInTheDocument();
   });
 
   it('renders the Release div with a text and version number and does not render when empty', function() {
-    const releaseEmpty = getShallowComponent().find('div.nx-global-sidebar__release'),
-        release = getShallowComponent({ releaseText: 'React Shared Components: 3.1.4' })
-            .find('div.nx-global-sidebar__release');
+    const { container: viewWithoutReleaseText} = quickRender(),
+        viewWithReleaseText = quickRender({ releaseText: 'React Shared Components: 3.1.4' });
+
+    expect(viewWithReleaseText.getByText('React Shared Components: 3.1.4')).toBeVisible();
 
-    expect(releaseEmpty).not.toExist();
-    expect(release).toHaveText('React Shared Components: 3.1.4');
+    expect(viewWithoutReleaseText.querySelector('.nx-global-sidebar__release')).not.toBeInTheDocument();
```

Comment:
This is, of course an impl detail, but I can see how it can be challenging to know how to test the negative case when it just doesn't add anything.  How about tests for each optional prop _by themselves_ which assert not only that the render includes the relevant text but that it includes _only_ that text (and anything else that's supposed to be there by default, if there is any).  And then aside from those tests of the props individually have one other test where they're all used together and we assert that the text from all of them is present.

Reply:
> This is, of course an impl detail, but I can see how it can be challenging to know how to test the negative case when it just doesn't add anything. How about tests for each optional prop _by themselves_ which assert not only that the render includes the relevant text but that it includes _only_ that text (and anything else that's supposed to be there by default, if there is any). And then aside from those tests of the props individually have one other test where they're all used together and we assert that the text from all of them is present.

Assuming that you suggested to take out the `querySelector` tests for negative cases, and only make organizational changes, I *think* I made changes based on your suggestions. 

Can you please check 4d234960d5a980fe8bfd11ddb392c5085d58bb8c to make sure we're on the same page?